lib/zuora/objects/amendment.rb

```diff
@@ -6,20 +6,20 @@ class Amendment < Base
     validates_length_of :name, :maximum => 100
     validates_inclusion_of :auto_renew, :in => [true, false], :allow_nil => true
     validates_length_of :code, :maximum => 50, :allow_nil => true
-    # validates_datetime_of :contract_effective_date, :allow_nil => true
-    # validates_datetime_of :customer_acceptance_date, :allow_nil => true
-    # validates_datetime_of :effective_date, :allow_nil => true
-    # validates_datetime_of :service_activation_date, :if => Proc.new { |a| a.status == 'PendingAcceptance' }
+    validates_datetime_of :contract_effective_date, :allow_nil => true
+    validates_datetime_of :customer_acceptance_date, :allow_nil => true
+    validates_datetime_of :effective_date, :allow_nil => true
+    validates_datetime_of :service_activation_date, :if => Proc.new { |a| a.status == 'PendingAcceptance' }
```

Comment:
Upon comparing `zuora.a.38.0.wsdl` and `zuora.a.78.0.wsdl`, it was found that fields previously validated with `validates_datetime_of` are no longer valid because their data type has changed from datetime to date.
For example 
# contract_effective_date https://github.com/Sage/zuora/blob/master/wsdl/zuora.a.38.0.wsdl#L657
https://github.com/Sage/zuora/blob/46b7ecfb2dce048c68e2445a637874284b558a5e/wsdl/zuora.a.78.0.wsdl#L953

The validation will have to be changed to
`validates_date_of :contract_effective_date, :allow_nil => true`

Reply:
Resolved here [04f281f](https://github.com/Sage/zuora/pull/18/commits/04f281f9a0a0b07961e1b1f4ebd5465beb8c27bc)